src/hooks/useElementIconTextures.ts

```diff
@@ -0,0 +1,30 @@
+import { getAssetPath } from "@/components/utils";
+import { useSRGBTexture } from "@/hooks/useSRGBTexture";
+import { Texture } from "three";
+import { useMemo } from "react";
+
+export type ElementName = "sun" | "temperature" | "nutrients" | "salinity" | "oxygen";
+
+const useElementIconTextures = (): {
+  [K in ElementName]: Texture;
+} => {
+  const textures = {
+    sun: useSRGBTexture(getAssetPath("element_icon", "sun")),
+    temperature: useSRGBTexture(getAssetPath("element_icon", "temperature")),
+    nutrients: useSRGBTexture(getAssetPath("element_icon", "nutrients")),
+    salinity: useSRGBTexture(getAssetPath("element_icon", "salinity")),
+    oxygen: useSRGBTexture(getAssetPath("element_icon", "oxygen")),
+  };
+
+  return useMemo(() => {
```

Comment:
I believe useMemo is unnecessary here, as far as I know the drei useTexture is already memoized